src/components/SearchFilters/SearchFilters.tsx

```diff
@@ -90,6 +90,12 @@ const SearchFilters = ({
   }
 
   const [focusedFilter, setFocusedFilter] = useState<string | null>(null)
+
+  // Only display Subject filter if there's a query term
+  if (router.query?.q === "" || !router.query.q) {
```

Comment:
would it makes sense to only remove these filters when there is no query and there are subject filters applied?